core/src/main/scripts/importer/validation_report_template.html.jinja

```diff
@@ -167,6 +166,60 @@
                 {% endif %}
                 <td>{{ record.getMessage()|e }}</td>
                 <td>{{ format_aggregated(record, 'cause') }}</td>
+                {% macro create_modal(filename, record, attr_name, num_record) %}
+                    {% if record[attr_name + '_list'] is defined -%}
+                      {% if (record['show_all_values'] is not defined or record['show_all_values'] == False) and (record[attr_name + '_list']|length) > 3 -%}
+                        <!-- Trigger/Open The Modal -->
+                        {% set num_values = record[attr_name + '_list']|length|string %}
+                        {% set modal_name = "modal" + filename.rsplit('.', 1)[0] + num_record + attr_name %}
+                        <a href="#showValues" data-toggle="modal" data-backdrop="false" data-target=".{{ modal_name }}">See all {{ num_values }} values </a>
+
+                        <div class="modal fade {{ modal_name }}" tabindex="-1" role="dialog" aria-labelledby="myLargeModalLabel">
+                          <div class="modal-dialog modal-lg" role="document">
+                            <div class="modal-content">
+                              <div class="modal-header">
+                                <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+                                <h4 class="modal-title" id="gridSystemModalLabel">Listing all values for the message: {{ record.getMessage() }}</h4>
+                              </div>
+                              <div class="modal-body">
+                                <table class="table">
+                                  <thead>
+                                    <tr>
+                                      <th>Line Number</th>
+                                      <th>Column Number</th>
+                                      <th>Value Encountered</th>
+                                    </tr>
+                                  </thead>
+                                  <tbody>
+                                    {% for value in record[attr_name + '_list'] %}
+                                      <tr>
+                                        <td>{% if value[0] is not none -%}
```

Comment:
It's been a long time since I wrote Jinja2 code, but these `if` blocks look a bit drawn out and repetitive. Do you think something like [inline if expressions](https://jinja.palletsprojects.com/en/2.11.x/templates/#if-expression) or a macro could make this block less cumbersome to read?

In Python even indexing tuples with integers like this is usually not the most readable way to do things, but I imagine that Jinja2 isn't as fancy in that regard, it's designed for templating and not for logic.

Reply:
As I said, too long ago, but I imagine that ‘undefined’ in the context of the [`default()`](https://jinja.palletsprojects.com/en/2.11.x/templates/#default) filter does not include ‘having a value of `none`’?

Reply:
Thanks again for suggesting the inline if - the code looks much better now. And as you mention, unfortunately the `default()` filter does not include `none` as an undefined value.